modular_skyrat/master_files/code/modules/client/preferences/antag_opt_in_preferences.dm

```diff
@@ -0,0 +1,91 @@
+/datum/config_entry/flag/disable_antag_opt_in_preferences
+	default = FALSE
+
+/datum/preference/toggle/master_antag_opt_in_preferences
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "master_antag_opt_in_pref"
+	savefile_identifier = PREFERENCE_PLAYER
+	default_value = TRUE
+
+/datum/preference/toggle/master_antag_opt_in_preferences/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+
+	return TRUE
+
+/datum/preference/toggle/master_antag_opt_in_preferences/deserialize(input, datum/preferences/preferences)
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+	. = ..()
+
+/datum/preference/toggle/antag_opt_in
+	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
+	savefile_identifier = PREFERENCE_PLAYER
+	savefile_key = "antag_opt_in_pref"
+	default_value = TRUE
+
+/datum/preference/toggle/antag_opt_in/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+
+	return preferences.read_preference(/datum/preference/toggle/master_antag_opt_in_preferences)
+
+/datum/preference/toggle/antag_opt_in/deserialize(input, datum/preferences/preferences)
+	if(CONFIG_GET(flag/disable_antag_opt_in_preferences))
+		return FALSE
+	if(!preferences.read_preference(/datum/preference/toggle/master_antag_opt_in_preferences))
+		return FALSE
+	. = ..()
```

Comment:
```suggestion
	return ..()
```
general rule of thumb, if youre returning on the line after you do . = ..(), just do return ..(), its enforced by most maintainers and is good style

Reply:
. is the default return value of the proc, and when the proc ends it always returns . unless specified otherwise

```
/proc/test()
    . = "a"
```
 this returns "a"

```
/proc/test()
    . = "a"
    return "b"
``` 

this returns "b"